roll/__init__.py

```diff
@@ -244,13 +248,13 @@ def __init__(self):
 
     async def __call__(self, request: Request, response: Response):
         try:
-            request.routing, params = self.routes.match(request.path)
+            request.route = Route(*self.routes.match(request.path))
             if not await self.hook('request', request, response):
-                if request.method not in request.routing:
+                if request.method not in request.route.payload:
                     raise HttpError(HTTPStatus.METHOD_NOT_ALLOWED)
-                request.kwargs.update(params)
-                handler = request.routing[request.method]
-                await handler(request, response, **params)
+                request.kwargs.update(request.route.vars)
```

Comment:
You can drop `request.kwargs` at this point :)

Reply:
And renaming it to `extras` in slots for extra usage? 🙃

Reply:
Nah! :)
You really want to do `request['user']` or `request['session']` not `request.extras['user']`… ;)
I'll push something in that direction tomorrow :)